service/feeder/feeder.go

```diff
@@ -38,25 +38,22 @@ func FromWAL(reader WALReader) *Feeder {
 	return &f
 }
 
-// Update returns the next trie update.
-func (f *Feeder) Update() (*ledger.TrieUpdate, error) {
+// Updates returns all updates in wal.
+func (f *Feeder) Updates() ([]*ledger.TrieUpdate, error) {
 
 	// We read in a loop because the WAL contains entries that are not trie
 	// updates; we don't really need to care about them, so we can just skip
 	// them until we find a trie update.
-	for {
+	updates := make([]*ledger.TrieUpdate, 1000)
```

Comment:
this seems like a bug.  you want `make([]*ledger.TrieUpdate, 0, 1000)` instead?

Reply:
yea, this needs `len=0` given the `append` on line 81